lib/removeUnusedElements.js

```diff
@@ -158,6 +166,62 @@ Remove.bufferView = function(gltf, bufferViewId) {
     }
 };
 
+Remove.mesh = function(gltf, meshId) {
+    const meshes = gltf.meshes;
+    meshes.splice(meshId, 1);
+
+    ForEach.node(gltf, function(n) {
+        if(defined(n.mesh)) {
+            if(n.mesh > meshId) {
+                --n.mesh;
+            } else if(n.mesh === meshId) {
+                /* Remove reference to deleted mesh */
+                delete n.mesh;
+            }
+        }
+    });
+};
+
+Remove.node = function(gltf, nodeId) {
+    const nodes = gltf.nodes;
+
+    /* Remove this node from parent */
+    const parentId = findParentNode(gltf, nodeId);
+    if(parentId >= 0) {
+        nodes[parentId].children = nodes[parentId].children.filter(x => x !== nodeId);
+    }
+
+    nodes.splice(nodeId, 1);
+
+    /* Shift all node references */
```

Comment:
Also shift node references for skeletons, joints, animations, and `KHR_technique_webgl`.